File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
class Solution {
    public int solution(String my_string) {
        String temp = "";
        int ret = 0;
        for(char c : my_string.toCharArray()){
            if(Character.isAlphabetic(c) && temp.length() > 0) {
                ret += Integer.parseInt(temp);
                temp = "";
            } else if(!Character.isAlphabetic(c)) temp += c;
        }
        if(temp.length() != 0) ret += Integer.parseInt(temp);
        return ret;
    }
}

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.py
def spplit(s):
    nums = [str(i) for i in range(10)]
    temp = ''
    ret = []
    for c in s:
        if c not in nums:
            if temp:
                ret.append(int(temp))
            temp = ''
            continue
        else: temp += c
    if temp: ret.append(int(temp))
    return ret

def solution(my_string):
    ints = spplit(my_string)
    return sum(int(x) for x in ints)    